### PR TITLE
Add Artillery Genius Pro config

### PR DESCRIPTION
**Pull request by jayofelony**
A basic config for the Artillery Genius Pro.

Signed-off-by: Jeroen Oudshoorn <oudshoorn.jeroen@gmail.com>

### CODE REVIEW

**Comment by JamesH1978** [comment]
Thank you for submitting a PR, as mentioned in the one you just closed, please refer to point 3 in "What to expect in a review" in https://github.com/Klipper3d/klipper/blob/master/docs/CONTRIBUTING.md and provide a signed off by line.

Also please refer to point 4 in the guidelines in the following link and add the printers.test with your new file for regression testing https://github.com/Klipper3d/klipper/blob/master/docs/Example_Configs.md

Thanks
James

**Comment by jayofelony** [comment]
Yes I closed it as my cfg file was gone somehow from my clone. I edited the first commit message, updated the printers.test file. I hope the checks all are good.

File: config/printer-artillery-genius-pro-2022.cfg [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+microsteps: 16
+rotation_distance: 7.1910
+nozzle_diameter: 0.400
+filament_diameter: 1.750
+heater_pin: PC9
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PC0
+min_temp: 0
+max_temp: 250
+pressure_advance = 0.088
```
Comment (JamesH1978) [inline config/printer-artillery-genius-pro-2022.cfg:27]: Please don't set a PA value in a stock config, this is for the end user to tune

Thanks
James

File: config/printer-artillery-genius-pro-2022.cfg [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+control: pid
+pid_Kp: 23.223
+pid_Ki: 1.518
+pid_Kd: 88.826
+
+[input_shaper]
```
Comment (JamesH1978) [inline config/printer-artillery-genius-pro-2022.cfg:33]: Please don't set IS values in a stock config, this is for the end user to tune

Thanks
James

File: config/printer-artillery-genius-pro-2022.cfg [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+[bed_screws]
+screw1: 38,45
+screw2: 180,45
+screw3: 180,180
+screw4: 38,180
+
+[fan]
+pin: PC6
+pin: PC8
```
Comment (JamesH1978) [inline config/printer-artillery-genius-pro-2022.cfg:89]: you cannot have 2 pin: in [fan] section

Thanks
James

File: config/printer-artillery-genius-pro-2022.cfg [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+[heater_fan hotend_fan]
+pin: PC7
+heater: extruder
+heater_temp: 50.0
+
+[controller_fan stepper_fan]
+pin: PC6
+idle_timeout: 300
+
+[mcu]
```
Comment (JamesH1978) [inline config/printer-artillery-genius-pro-2022.cfg:100]: is there any reason why you are not using the by-id, the end user may not have the same naming convention in their host OS

Thanks
James

File: config/printer-artillery-genius-pro-2022.cfg [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+[printer]
+kinematics: cartesian
+max_velocity: 500
+max_accel: 4000
+max_z_velocity: 50
+square_corner_velocity: 5.0
+max_z_accel: 100
+
+[bltouch]
+sensor_pin: PC2
```
Comment (JamesH1978) [inline config/printer-artillery-genius-pro-2022.cfg:112]: Touch devices usually need a pull up ^ on the sensor_pin.

Thanks
James

File: config/printer-artillery-genius-pro-2022.cfg [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+algorithm: bicubic
+move_check_distance: 3.0
+
+[safe_z_home]
+home_xy_position: 110,110
+speed: 100
+z_hop: 10
+z_hop_speed: 5
+
+[virtual_sdcard]
```
Comment (JamesH1978) [inline config/printer-artillery-genius-pro-2022.cfg:135]: This is also for the end user to set, as it may not be this address for their use.

Thanks
James

**Review by JamesH1978 — CHANGES_REQUESTED** [review]
Please see my comments.

Thanks
James

File: config/printer-artillery-genius-pro-2022.cfg (hunk 5 above)
Comment (jayofelony) [reply config/printer-artillery-genius-pro-2022.cfg:106]: This one does not, same are the Artillery Sidewinder.

File: config/printer-artillery-genius-pro-2022.cfg [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+# To flash this firmware, set the physical bridge between +3.3V and Boot0 PIN
+# on Artillery_Ruby mainboard. Then run the command:
+#   make flash FLASH_DEVICE=/dev/serial/by-id/usb-Klipper_stm32f401xc_*-if00
+
+# See docs/Config_Reference.md for a description of parameters.
+
+[extruder]
+max_extrude_only_distance: 700.0
+max_extrude_cross_section: 50.0
```
Comment (JamesH1978) [inline config/printer-artillery-genius-pro-2022.cfg:14]: This should not be set at 50, this is a safety feature, the line should be left out.

Thanks
James

**Comment by JamesH1978** [comment]
bar that one last comment about the cross section safety and a read through again, i think we should be good

Thanks
James

**Comment by jayofelony** [comment]
All done. Thanks for the help getting the PR done correctly.

**Review by JamesH1978 — APPROVED** [review]
Looks fine to me now, thanks

James

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Pull request merged**